### PR TITLE
Laboratory work #2, Beryozkin Daniil - 21FPL1

### PR DESCRIPTION
**Pull request by danind**

### CODE REVIEW

File: lab_1_keywords_tfidf/main.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 from math import log
 
 
-def check_list(user_input: Any, elements_type: type, can_be_empty: bool) -> bool:
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:9]: You can't change the first lab

File: lab_2_keywords_cooccurrence/main.py [hunk 2]
```diff
@@ -55,7 +92,21 @@ def calculate_word_degrees(candidate_keyword_phrases: KeyPhrases,
...
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not (check_list(candidate_keyword_phrases, tuple) and check_list(content_words, str)):
+        return None
+    word_degrees_dict = {}
+    for phrase in candidate_keyword_phrases:
+        for word in phrase:
+            if word in word_degrees_dict:
+                if word in content_words:
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:101]: You can take that `if` out of this statement and out of `else` - this is common part.

**Review by artyomtugaryov — COMMENTED** [review]
Good. I'm sure you can do the job for 8 and 10 points

**Comment by danind** [comment]
My code can pass all test, but it cannot process all texts in start.py
I guess, there are problems with function «extract_candidate_keyword_phrases_with_adjoining», but I can't solve them :(

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
You need to fix start.py for green CI

**Review by artyomtugaryov — APPROVED** [review]

**Pull request closed without merging**